Zend/zend_attributes.c

```diff
@@ -84,6 +84,11 @@ static void validate_allow_dynamic_properties(
 			ZSTR_VAL(scope->name)
 		);
 	}
+	if (scope->ce_flags & ZEND_ACC_ENUM) {
```

Comment:
The change looks good but I'm wondering whether it isn't an oversight of the enum RFC that attributes targeting classes are allowed for enums? It looks like to me that we should(n't have) allow(ed) this, so the best course of action would be to add a new target for enums, and only allow the usage if it's set.

cc @iluuu1994

Reply:
[[It's not an oversight, ]()](https://wiki.php.net/rfc/enumerations)

> Enums and cases may have attributes attached to them, like any other language construct. The TARGET_CLASS target filter will include Enums themselves.

Reply:
OK, I see it now.

Reply:
@kocsismate Do you have a special use-case for TARGET_ENUM? We considered it back then but it never came up again, so I assume use-cases are limited.

Reply:
> Do you have a special use-case for TARGET_ENUM? We considered it back then but it never came up again, so I assume use-cases are limited.

No, I don't have any... I just proposed this in order to be able to retain the original behavior if we prevented the usage of attributes targeting classes for enums. But I realized that it would already be a possibly serious BC break since userland attributes have already been relying on the current behavior. Sorry for the noise!